src/plugins/workflow_designer/src/cmdline/WorkflowCMDLineTasks.cpp

```diff
@@ -55,44 +55,44 @@ WorkflowRunFromCMDLineBase::WorkflowRunFromCMDLineBase()
       workflowRunTask(NULL) {
     GCOUNTER(cvar, "workflow_run_from_cmdline");
 
-    CMDLineRegistry *cmdLineRegistry = AppContext::getCMDLineRegistry();
-
-    // try to process schema without 'task' option (it can only be the first one)
-    QStringList pureValues = CMDLineRegistryUtils::getPureValues();
-    if (!pureValues.isEmpty()) {
-        QString schemaName = pureValues.first();
-        processLoadSchemaTask(schemaName, 1);    // because after program name
-    }
-    if (loadTask != NULL) {
-        addSubTask(loadTask);
-        return;
-    }
-
-    // process schema with 'task' option
-    int taskOptionIdx = CMDLineRegistryUtils::getParameterIndex(WorkflowDesignerPlugin::RUN_WORKFLOW);
-    if (taskOptionIdx != -1) {
-        processLoadSchemaTask(cmdLineRegistry->getParameterValue(WorkflowDesignerPlugin::RUN_WORKFLOW, taskOptionIdx), taskOptionIdx);
-    }
-    if (loadTask == NULL) {
-        setError(tr("no task to run"));
-        return;
-    }
-    addSubTask(loadTask);
+    connect(this, &Task::si_stateChanged, this, &WorkflowRunFromCMDLineBase::sl_stateChanged);
```

Comment:
Is it a bug here: this connects to this? We have a dedicated state methods like prepare/run/report for each state change.

Reply:
It's not a bug. the `report` function is for the report generation, not for some post-processing stuff. It's apples and oranges, no need of mixing them. Also, we use this way of connection very often, look, for example, `BaseOneOneWorker` (54), `BaseOneOneWorker` (62), `WorkflowMonitor`(70) and a lot of (actually, A LOT OF) other similar places (search for using `si_stateChanged`).

Reply:
> actually, A LOT OF

Could you please point some places where a task uses si_stateChanged signal to get notification about its own state updates?
I checked your examples and they look like this:
```
connect(lastTickTask, SIGNAL(si_stateChanged()), SLOT(sl_taskFinished()));
```
Note: that the sender and the recipient are different objects.
In your case you have 2 'this' in the connect line.

Reply:
OK, you're right, your suggestions?